src/Rules/SingleUseCodeExist.php

```diff
@@ -2,38 +2,28 @@
 
 namespace Enjin\Platform\Beam\Rules;
 
+use Closure;
 use Enjin\Platform\Beam\Models\BeamClaim;
 use Enjin\Platform\Beam\Services\BeamService;
-use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Contracts\Validation\ValidationRule;
 
-class SingleUseCodeExist implements Rule
+class SingleUseCodeExist implements ValidationRule
 {
     /**
      * Determine if the validation rule passes.
      *
      * @param string $attribute
      * @param mixed  $value
+     * @param Closure $fail
      *
-     * @return bool
+     * @return void
      */
-    public function passes($attribute, $value)
+    public function validate(string $attribute, mixed $value, Closure $fail): void
     {
-        if (BeamService::isSingleUse($value)) {
-            return BeamClaim::withSingleUseCode($value)
-                ->claimable()
-                ->first();
+        if (BeamService::isSingleUse($value) && null !== BeamClaim::withSingleUseCode($value)->claimable()->first()) {
```

Comment:
I think we can just replace this with exists
```suggestion
        if (BeamService::isSingleUse($value) && BeamClaim::withSingleUseCode($value)->claimable()->exists()) {
```